### PR TITLE
#165936718 - Feature For Author Notifications

### PR DESCRIPTION
**Pull request by ZackAtama**
#### What does this PR do?
- Adds a feature to notify an author when a user follows them.
- Adds a feature to notify an author when a user likes their article.
- Adds a feature to notify an author when a user favorites their article.

#### Description of the task to be completed?
- When a user follows an author, the author receives a notification.
- When a user likes an article, the author of that article receives a notification.
- When a user favorites an article, the author of that article receives a notification.

#### What are the relevant pivotal tracker stories
- [#165936718](https://www.pivotaltracker.com/story/show/165936718)

### CODE REVIEW

**Review by nadralia — APPROVED** [review]
Nice work @AtamaZack

File: authors/apps/notifications/models.py [hunk 1]
```diff
@@ -87,7 +88,62 @@ def save_notification(sender, instance, **kwargs):
...
+            instance.favorited_by.username
+        ) 
+        link = "{}/api/articles/{}/".format(settings.WEB_HOST, article.id)
+
+    if isinstance(instance, AuthorFollowing):
+        receiver = instance.following
+        notification_msg = 'You have been followed by {}'.format(instance.follower)
+        link = "{}/api/profiles/{}/".format(
+            settings.WEB_HOST,
+            instance.following.username
```
Comment (kalsmic) [inline authors/apps/notifications/models.py]: well done @AtamaZack, I have tested out the three instances for receiving notifications that you specified in the PR description and they work fine. However, I would think that the view button should refer back to the follower.
Screenshot 2019-05-15 at 09 01 51

**Review by kalsmic — CHANGES_REQUESTED** [review]

File: authors/apps/notifications/models.py (hunk 1 above)
Comment (ZackAtama) [reply authors/apps/notifications/models.py]: @kalsmic, thanks for the feedback, resolved!

**Review by bisonlou — APPROVED** [review]
Well done @AtamaZack

**Review by kalsmic — APPROVED** [review]

**Review by llwasampijja — APPROVED** [review]
nice work @AtamaZack

**Pull request merged**